src/main.ts

```diff
@@ -1,8 +1,22 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
+import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
-  await app.listen(3000);
+  app.setGlobalPrefix('api');
+
+  const options = new DocumentBuilder()
+  .setTitle('Online Shop API')
+  .setDescription("API for an online shop")
+  .setVersion('1.0.0')
+  .build();
+
+  const document = SwaggerModule.createDocument(app, options);
+  SwaggerModule.setup('api', app, document);
+
+  const port = process.env.PORT || 3000
+  await app.listen(port);
```

Comment:
**issue:** note you have added class-validator package and annotated the various dto with the appropiate validations, but you need to add the validator pipe globally here:
```typescript
  app.useGlobalPipes(new ValidationPipe());
```
Link: https://docs.nestjs.com/techniques/validation#auto-validation

Reply:
Solved.
I missed that indeed, and I actually had a suspicion that those decorators for validation weren't actually working.
Thank you for pointing it out!